test/TestAssets/OutOfProcDataCollector/SampleDataCollector.cs

```diff
@@ -41,6 +42,9 @@ private void Events_TestCaseEnd(object sender, TestCaseEndEventArgs e)
         private void Events_TestCaseStart(object sender, TestCaseStartEventArgs e)
         {
             this.logger.LogWarning(this.context.SessionDataCollectionContext, "TestCaseStarted " + e.TestCaseName);
+            var filename = Path.Combine(AppContext.BaseDirectory, "testcasefilename" + i++ + ".txt");
```

Comment:
`this.dataCollectionSink.SendFileAsync(e.Context, filename, true);`
The last parameter is deleteFile, which deletes the temporary file after moving it to test results directory.